commands/reserves/reservableitem.js

```diff
@@ -0,0 +1,70 @@
+const { Op } = require("sequelize");
+
+// Add a new reservable item to the raid db
+exports.run = async function(client, message, args) {
+
+    if (!client.permission.manageReserves(message.member)) {
+		return message.channel.send('**Error:** You need permission to be able to add items to the reservable items list.');
+	}
+
+
+    if (!args[0]) {
+        return message.channel.send('**Error:** Correct usage is ``+reservableitem add 30024 TK Mantle of the Elven Kings``');
+    };
+
+
+    let addOrRemove = args.shift().toLowerCase();
+
+    if (addOrRemove !== 'add' && addOrRemove !== 'remove') {
+        return message.channel.send('**Error:** Must specify **ADD** or **REMOVE**!')
+    };
+
+
+    let record = {
+        'itemID': parseInt(args.shift()),
+        'raid': args.shift().toUpperCase(),
+        'name': args.join(' ')
+    };
+
+
+    let validRaidList = ['MC', 'BWL', 'Ony', 'ZG', 'AQ20', 'AQ40', 'Naxx', 'GL', 'Kara', 'ML', 'SSC', 'TK'];
+
+    if (!validRaidList.includes(raid)) {
+        return message.channel.send('**Error:** Must be a valid raid ID (MC, BWL, Ony, ZG, AQ20, AQ40, Naxx, GL, Kara, ML, SSC, TK)')
+    }
+
+
+    if (isNaN(record.itemID)) {
+        return message.channel.send('**Error:** Invalid item number!');
+    };
+
+
+    if (addOrRemove === "add") {
+        // Check if item name or item number already exists under that raid name
+        client.models.reserveItem.findOne({
+                where: {
+                    [Op.and]: [
+                    {   raid: record.raid,
+                        [Op.or]: [
+                            {name: record.name}, {itemID: record.itemID}

```

Comment:
There are a couple of strange cases where there can be two items with the same name and different Item IDs, such as the head of onyxia, as it starts a different quest for each faction.  It'd probably be best to not fail on duplicate name, only duplicate ID.